open-banking-accelerator/components/com.wso2.openbanking.accelerator.gateway/src/main/java/com/wso2/openbanking/accelerator/gateway/executor/dcr/DCRExecutor.java

```diff
@@ -610,6 +250,409 @@ public void postProcessRequest(OBAPIRequestContext obapiRequestContext) {
+                    GatewayConstants.BEARER_TAG.concat(token), "", "");
+            if (publishedAPIsResponse == null) {
+                log.error("Error while retrieving published APIs");
+                //delete SP created to call dev portal REST APIs
+                callDelete(urlMap.get(GatewayConstants.IAM_DCR_URL).toString().concat("/")
+                        .concat(clientId), basicAuthHeader);
+                handleInternalServerError(obapiResponseContext, OpenBankingErrorCodes.REGISTRATION_UPDATE_ERROR);
+                return;
+            }
+            List<String> apiIDList = new ArrayList<>();
+            if (StringUtils.isEmpty(softwareStatement)) {
+                filterRegulatoryAPIs(regulatoryAPIs, publishedAPIsResponse.getAsJsonObject()
+                        .get(OpenBankingConstants.API_LIST).getAsJsonArray(), Collections.emptyList());
+            } else {
+                filterRegulatoryAPIs(regulatoryAPIs, publishedAPIsResponse.getAsJsonObject()
+                        .get(OpenBankingConstants.API_LIST).getAsJsonArray(), getRolesFromSSA(softwareStatement));
+            }
+
+            List<String> newApisListToSubscribe = getNewAPIsToSubscribe(apiIDList, subscribedAPIIdList);
+            if (!newApisListToSubscribe.isEmpty()) {
+                JsonArray subscribeAPIsPayload = getAPISubscriptionPayload(applicationId, newApisListToSubscribe);
+                JsonElement subscribeAPIsResponse = callPost(urlMap.get(GatewayConstants.API_SUBSCRIBE_URL)
+                        .toString(), subscribeAPIsPayload.toString(), "Bearer ".concat(token));
+                if (subscribeAPIsResponse == null) {
+                    log.error("Error while subscribing to APIs");
+                    //delete SP created to call dev portal REST APIs
+                    callDelete(urlMap.get(GatewayConstants.IAM_DCR_URL).toString().concat("/")
+                            .concat(clientId), basicAuthHeader);
+                    handleInternalServerError(obapiResponseContext,
+                            OpenBankingErrorCodes.REGISTRATION_UPDATE_ERROR);
+                    return;
+                }
+            }
+            //delete IAM application used to invoke am rest endpoints
+            if (!callDelete(urlMap.get(GatewayConstants.IAM_DCR_URL).toString().concat("/")
+                    .concat(clientId), basicAuthHeader)) {
+                handleInternalServerError(obapiResponseContext, OpenBankingErrorCodes.REGISTRATION_UPDATE_ERROR);
+            }
+        } catch (ParseException | IOException | URISyntaxException | OpenBankingException e) {
+            log.error("Error occurred while creating application", e);
+            handleInternalServerError(obapiResponseContext, OpenBankingErrorCodes.REGISTRATION_UPDATE_ERROR);
+        }
+    }
+
+    private void processDeleteRequestResponse(OBAPIResponseContext obapiResponseContext, String basicAuthHeader) {
```

Comment:
add method comment

Reply:
Resolved in [8657c32](https://github.com/wso2/financial-open-banking/pull/62/commits/8657c32a1adada4bcf5b652287b825b09a36ca96)